parsel/selector.py

```diff
@@ -189,7 +189,9 @@ def __init__(self, text=None, type=None, namespaces=None, root=None,
 
         if text is not None:
             if not isinstance(text, six.text_type):
-                raise TypeError("text argument should be of type %s" % six.text_type)
+                msg = "text argument should be of type %s, got %s" % (
+                           six.text_type, text.__class__)
```

Comment:
I'm unsure now - is it better to show `repr(value)`?

Reply:
No, if the problem is the type I think it makes sense to output the type, and I think that’s the usual approach of the standard library.